staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -41,10 +42,13 @@ type serverHandlerConfig struct {
 	statusCode       int
 	connectionHeader string
 	upgradeHeader    string
+	delay            time.Duration
 }
 
 func serverHandler(t *testing.T, config serverHandlerConfig) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
+		time.Sleep(config.delay)
```

Comment:
modifying the existing this way doesn't look great, I can see you can make it as if config.delay then sleep, but if we going to do this we may want to hold the connection in different stages ... maybe just adding a test for this the same way the other contributor did

https://github.com/kubernetes/kubernetes/pull/109651

but is better to wait for more opinions about this change, I don't want to suggest things and then it turns out the solution is not accepted

Reply:
I had doubts, too. I can happily extract the new test into a separate one. I still did it this way to avoid code duplication. Moreover, the added `time.Sleep` is a no-op for test cases not setting `config.delay`.

Reply:
let's go with separate case then,

Reply:
I extracted the test case. I'll take care of everything else after my vacation next week :)

Reply:
no worries, we'll have to wait for 1.28 for this, so don't need to rush it